File: input_type.cpp
#include <iostream>
// #include <string>
// #include <algorithm>
std::string input_type(std::string value) { 
    if(std::isalpha(value[0])) {
        return "string";
    } 
    else if(std::isdigit(value[0])) {
        if(value.find('.') != std::string::npos)
        return "double";
        return "integer";
    } 
    return "unknown";
}

int main()
{
    // std::string str = "Hello, World!";
    // std::string str = "12345";
    std::string str = "12.345";
    std::cout << input_type(str) << std::endl;
    return 0;
}

File: raytest/inputtype.py
# from typing import List
# def input_type(value: str) -> str:
#     if value.isalpha():
#         return "string"
#     elif value[0].isdigit():
#         if(value.find('.')):
#             return "double"
#         else:
#             return "integer"
#     return value

class test:
    def __init__(self,name,id):
        self.name = name
        self.id = id
    def talk(self):
        print("my name ",self.name,"my id ",self.id)
        print(self)

opj = test(name="eslam",id=1112)
print(opj.name)
opj.talk()

class pp:
    def __init__(self,n,w,h):
        self.name = n
        self.whgit = w
        self.hight = h
        print("name ",n,"whgit ",w,"hight ",h)
    def tosum(a,b):
        print(a+b)

p = pp("eslam",55,173)

pp.tosum(10,10)

print(dir(p))